iceberg-rust/src/table/manifest_list.rs

```diff
@@ -683,42 +687,38 @@ impl<'schema, 'metadata> ManifestListWriter<'schema, 'metadata> {
         )?;
 
         let mut manifest_writer = if let (Some(mut manifest), Some(manifest_bytes)) =
-            (self.selected_manifest, selected_manifest_bytes_opt)
+            (self.selected_manifest.take(), selected_manifest_bytes_opt)
         {
             let manifest_bytes = manifest_bytes.await??;
 
             manifest.manifest_path =
-                new_manifest_location(&self.table_metadata.location, &self.commit_uuid, 0);
+                new_manifest_location(&self.table_metadata.location, &self.commit_uuid, content as usize);
```

Comment:
This is so that data and delete manifest path is different, otherwise if both are present the deletes will overwrite the data (as they're applied second). 

This also needs fixing in the `append_multiple_filtered`, which could be merged with `append_filtered`; need to think about it some more.

Reply:
Ok, i solved this by having a per-instance manifest counter, which then helps to determine the next manifest id to use in the file name.